org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapterContext.java

```diff
@@ -147,4 +152,9 @@ public IVariableFormatter getVariableFormatter() {
     public void setVariableFormatter(IVariableFormatter variableFormatter) {
         this.variableFormatter = variableFormatter;
     }
+
+    @Override
+    public Map<String, String> getSourceLookupCache() {
+        return Collections.synchronizedMap(this.sourceMappingCache);

```

Comment:
Collections.synchronizedMap will create a new wrapper every time. should move it to `Line 24`

Reply:
Done